src/Core/Settings.h

```diff
@@ -169,6 +169,7 @@ class IColumn;
     M(Int64, os_thread_priority, 0, "If non zero - set corresponding 'nice' value for query processing threads. Can be used to adjust query priority for OS scheduler.", 0) \
     \
     M(Bool, log_queries, 1, "Log requests and write the log to the system table.", 0) \
+    M(Bool, log_formatted_queries, 0, "Log formatted queries and write the log to the system table.", 0) \
```

Comment:
But if formatted query stored into separate column, does option required?

Reply:
I think it's needed as the additional column might double the size of system.query_log.

Reply:
There are already lots of columns (i.e. `used_*`) that does not controlled by any setting.
Adding new setting just to control which columns will go to query_log just too much from my point of view (since this increases settings matrix).

I think that better alternative will be to add this new column as-is.

And later some setting to control which columns will go to query_log can be added.
i.e. `log_queries_columns=query,...|_compact`, where empty value means all columns `_compact` means some subset of basic columns and also user can set explicitly which columns does he want.
Although I'm not sure that it worth it...

Reply:
Actually we can extend the `log_queries` setting somehow. I'm not sure if it'll break compatibility

Reply:
bool -> uint64 is backward compatible, but bool -> string is not.

Reply:
I think current solution is Ok.